packages/core/src/connectivity/scp-cf/destination/destination.ts

```diff
@@ -55,7 +55,54 @@ export function parseDestination(
     }
   );
 
-  return sanitizeDestination(destination);
+  const additionalHeadersAndQueryParameters =
+    getAdditionalHeadersAndQueryParameters(destinationConfig);
+
+  return sanitizeDestination({
+    ...destination,
+    ...additionalHeadersAndQueryParameters
+  });
+}
+
+function getAdditionalProperties(
+  destinationConfig: DestinationConfiguration,
+  originalKeyPrefix: string
+): Record<string, any> {
+  const relevantConfigEntries = Object.entries(destinationConfig).filter(
+    ([key]) => key.startsWith(originalKeyPrefix)
+  );
+
+  return relevantConfigEntries.reduce(
+    (additionalProperties: Record<string, any>, [originalKey, value]) => {
+      const headerKey = originalKey.replace(originalKeyPrefix, '');
+      additionalProperties[headerKey] = value;
+      return additionalProperties;
+    },
+    {}
+  );
+}
+
+function getAdditionalHeadersAndQueryParameters(
```

Comment:
I find the code much clearer now. Anyway could you perhaps add a little doc about the situation in the kind of "There is a convention of putting parameters with XYZ on destination, client should handle bla bla. Link to the destination service docs" so that it is 100% clear what this is about.

Reply:
Makes sense!